Calibration/HcalAlCaRecoProducers/src/AlCaHOCalibProducer.cc

```diff
@@ -289,80 +255,39 @@ class AlCaHOCalibProducer : public edm::EDProducer {
 //
 // constructors and destructor
 //
-AlCaHOCalibProducer::AlCaHOCalibProducer(const edm::ParameterSet& iConfig)
-
-{
-   //register your products
+AlCaHOCalibProducer::AlCaHOCalibProducer(const edm::ParameterSet& iConfig) {
+  //register your products
 
-  theRootFileName = iConfig.getUntrackedParameter<string>("RootFileName","tmp.root");
-  m_digiInput = iConfig.getUntrackedParameter<bool>("digiInput", true);
+  theRootFileName = iConfig.getUntrackedParameter<std::string>("RootFileName","tmp.root");
   m_hbinfo = iConfig.getUntrackedParameter<bool>("hbinfo", false);
-  m_startTS = iConfig.getUntrackedParameter<int>("firstTS", 4);
 
-  m_hotime = iConfig.getUntrackedParameter<bool>("hotime", false);
-
-  if(m_startTS<0) m_startTS=0;
-  m_endTS = iConfig.getUntrackedParameter<int>("lastTS", 7);
-  if (m_endTS < m_startTS) m_endTS = m_startTS + 3;
-  if (m_endTS >9) m_endTS=9;
-  m_magscale = iConfig.getUntrackedParameter<double>("m_scale", 4.0);
-  m_sigma = iConfig.getUntrackedParameter<double>("sigma", 1.0);
+  m_sigma = iConfig.getUntrackedParameter<double>("sigma", 0.05);
 
   // keep InputTag muonTags_ since it is used below. - cowden
   muonTags_ =   iConfig.getUntrackedParameter<edm::InputTag>("muons");
+#ifdef COSMIC
   tok_muons_ = consumes<reco::TrackCollection>(muonTags_);
+#else
+  tok_muons_ = consumes<edm::View<reco::Muon> >(muonTags_);
+  tok_vertex_ = consumes<reco::VertexCollection >(iConfig.getParameter<edm::InputTag>("vertexTags"));
+  //  tok_lumi_ = consumes<LumiDetails ,edm::InLumi>(iConfig.getParameter<edm::InputTag>("lumiTags"));
+  tok_lumi_ = consumes<LumiScalersCollection>(iConfig.getParameter<edm::InputTag>("lumiTags"));
+#endif
   tok_ho_ = consumes<HORecHitCollection>(iConfig.getParameter<edm::InputTag>("hoInput"));
   tok_hbhe_ = consumes<HBHERecHitCollection>(iConfig.getParameter<edm::InputTag>("hbheInput"));
   tok_tower_ = consumes<CaloTowerCollection>(iConfig.getParameter<edm::InputTag>("towerInput"));  
 
-  // Since these accesses are currently commented out, I put the registration as "mayConsume". - cowden
-  tok_hlt_  = mayConsume<edm::TriggerResults>(iConfig.getParameter<edm::InputTag>("hltInput"));
-  tok_l1_ = mayConsume<L1GlobalTriggerReadoutRecord>(iConfig.getParameter<edm::InputTag>("l1Input"));
-  
   produces<HOCalibVariableCollection>("HOCalibVariableCollection").setBranchAlias("HOCalibVariableCollection");
   
   
-  if (m_hotime) {
+  if (m_occupancy) {
     edm::Service<TFileService> fs;
```

Comment:
I see here that you use `TFileService`, and plain ROOT files, but also you have an ALCARECO DQM sequence.
Assuming that you want to check the stability of HO calibration during data taking, I suggest to use plain DQM everywhere. Less code maintenance and more effective plots :-)